src/Listing/Bid/Bid.model.js

```diff
@@ -81,12 +106,77 @@ export class Bid extends Model {
     // Re-validate bids for assets with its fingerprint back to the original value
     await this.db.query(
       SQL`UPDATE ${raw(this.tableName)}
-        SET status = ${LISTING_STATUS.open}
+        SET status = ${
+          LISTING_STATUS.open
+        }, block_time_updated_at = ${blockTime}, block_number = ${blockNumber}
         WHERE block_time_created_at <= ${blockTime}
           AND token_address = ${tokenAddress}
           AND token_id = ${tokenId}
           AND status = ${LISTING_STATUS.fingerprintChanged}
           AND fingerprint = ${fingerprint}`
     )
   }
+
+  static async updateAssetOwner(
+    tokenAddress,
+    assetId,
+    seller,
+    blockTime,
+    blockNumber
+  ) {
+    return this.db.query(
+      SQL`UPDATE ${raw(this.tableName)}
+        SET seller = ${seller}, block_time_updated_at = ${blockTime}, block_number = ${blockNumber}
+        WHERE block_time_created_at <= ${blockTime}
+          AND token_address = ${tokenAddress} 
+          AND asset_id = ${assetId}
+          AND status = ANY(${[
+            LISTING_STATUS.open,
+            LISTING_STATUS.fingerprintChanged
+          ]})`
+    )
+  }
+
+  static async findBiddedAssetsByStatus(filters) {
+    const { status, bidder /*, seller  */ } = filters
+    const Assets = new Listing.getListableAssets()
+
+    // @nacho TODO: temp code. move to estate.queries or asset
+    // prettier-ignore
+    const estateHasParcels = () => 
+     SQL`(bid.asset_type = ${LISTING_ASSET_TYPES.estate}
+      AND EXISTS(SELECT 1 FROM ${raw('parcels')} AS p WHERE p.estate_id = bid.asset_id)
+      OR bid.asset_type != ${LISTING_ASSET_TYPES.estate})`
+
+    const selectAssetsSQL = Assets.map(
+      ({ tableName }) => `row_to_json(${tableName}.*) as ${tableName}`
+    ).join(', ')
+
+    const joinAssetsSQL = Assets.map(
+      ({ tableName }) =>
+        `LEFT JOIN ${tableName} as ${tableName} ON ${tableName}.id = bid.asset_id`
+    ).join('\n')
+
+    let assets = await db.query(
+      SQL`SELECT row_to_json(bid.*) as bids, ${raw(selectAssetsSQL)}
+          FROM ${raw(this.tableName)} as bid
+          ${raw(joinAssetsSQL)}
+          WHERE bid.status = ${status}
+            AND bid.bidder = ${bidder}
+            AND ${estateHasParcels()}`
+    )
+
+    // Keep only the model that had a bid defined from the Assets list
+    assets = assets.map(asset => {
+      for (const Model of Assets) {
+        if (asset[Model.tableName] != null) {
+          Object.assign(asset, asset[Model.tableName])
+        }
+        delete asset[Model.tableName]
+      }
+      return asset
+    })
```

Comment:
This method looks dangerously similar to `Marketplace`'s `filterAll` do you think it'd be possible to extract some common logic? like the asset `map`ing.

We might need a new concept/abstraction